fplog/fplog.cpp

```diff
@@ -308,7 +307,7 @@ class FPLOG_API Fplog_Impl
 
         const char* get_facility()
         {
-            return facility_.c_str();
```

Comment:
`std::lock_guard<std::recursive_mutex> lock(mutex_);` is needed here because we accessing map that can be modified by another thread and potentially lead to data corruption.